src/data_management/wms_layers/WmsLayerForm.tsx

```diff
@@ -0,0 +1,357 @@
+          value={values.download_url as string}
+          valueChanged={handleInputChange}
+          clearInput={clearInput}
+          validated={true}
+          // errorMessage={minLength(3, values.name as string)}
+          triedToSubmit={triedToSubmit}
+        />
+        <TextInput
+          title={'Legend Url'}
+          name={'legend_url'}
+          placeholder={'http://example.com'}
+          value={values.legend_url as string}
+          valueChanged={handleInputChange}
+          clearInput={clearInput}
+          validated={true}
+          triedToSubmit={triedToSubmit}
+        />
+        <TextInput
+          title={'Get Feature Url'}
+          name={'get_feature_info_url'}
+          placeholder={'http://example.com'}
+          value={values.get_feature_info_url as string}
+          valueChanged={handleInputChange}
+          clearInput={clearInput}
+          validated={true}
+          triedToSubmit={triedToSubmit}
+        />
+        <CheckBox
+          title={'Tiled'}
+          name={'tiled'}
+          value={values.tiled as boolean}
+          valueChanged={bool => handleValueChange('tiled', bool)}
+        />
+        <MinMaxZoomField
+          name={'minMaxZoom'}
+          value={{
+            minZoom: values.min_zoom as number,
+            maxZoom: values.max_zoom as number,
+          }}
+          valueChanged={(value:MinMax) => { 
+            if (values.min_zoom !== value.minZoom) {
+              handleValueChange('min_zoom', value.minZoom);
+            } else {
+              handleValueChange('max_zoom', value.maxZoom);
+            }
+          }}
+          triedToSubmit={triedToSubmit}
+        />
+        <SpatialBoundsField
+           name={'spatial_bounds'}
+           // @ts-ignore
+           value={values.spatial_bounds}
+           valueChanged={(value:any) => handleValueChange('spatial_bounds', value)}
+           clearInput={clearInput}
+           triedToSubmit={triedToSubmit}
+           otherValues= {{
+            wmsLayerName: values.name + '',
+            wmsLayerSlug: values.slug + '',
+            wmsLayerUrl: values.wms_url + '',
+           }}
+           geoServerError={geoserverError}
+           showGeoServerError={()=>setGeoserverError(true)}
+        />
+        <TextArea
+          title={'Options (JSON)'}
+          name={'options'}
+          placeholder={'Enter valid JSON'}
+          value={values.options as string}
+          valueChanged={handleInputChange}
+          clearInput={clearInput}
+          validated={(()=>{
+            try{
+              JSON.parse(values.options as string)
+            } catch(e) {
+              return false;
+            }
+            return true;
+          })()}
+          errorMessage={(()=>{
+            try{
+              JSON.parse(values.options as string)
+            } catch(e) {
+              return "needs to be valid JSON";
+            }
+            return "";
+          })()}
```

Comment:
we can refactor it to become 1 validator function

Reply:
done